Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -293,10 +294,59 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     AnalysisDataService::Instance().remove(wsName);
   }
 
+  /// Test the Nexus geometry loader from file
+  void testExecNexusLOKI() {
+    LoadInstrument loaderLOKI;
+    TS_ASSERT_THROWS_NOTHING(loaderLOKI.initialize());
+
+    // create a workspace with some sample data
+    wsName = "LoadInstrumentTestLOKI";
+    Workspace_sptr ws =
+        WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
+    Workspace2D_sptr ws2D = boost::dynamic_pointer_cast<Workspace2D>(ws);
+
+    // put this workspace in the data service
+    TS_ASSERT_THROWS_NOTHING(AnalysisDataService::Instance().add(wsName, ws2D));
+
+    const std::string definitionFile = "LOKI_Definition.hdf5";
+    loaderLOKI.setPropertyValue("Filename", definitionFile);
+    loaderLOKI.setPropertyValue("Workspace", wsName);
+    loaderLOKI.setProperty("RewriteSpectraMap", OptionalBool(true));
+    inputFile = loaderLOKI.getPropertyValue("Filename");
+
+    std::string result;
+    TS_ASSERT_THROWS_NOTHING(result = loaderLOKI.getPropertyValue("Filename"));
+    TS_ASSERT(!result.compare(inputFile));
+
+    TS_ASSERT_THROWS_NOTHING(result = loaderLOKI.getPropertyValue("Workspace"));
+    TS_ASSERT(!result.compare(wsName));
+
+    TS_ASSERT_THROWS_NOTHING(loaderLOKI.execute());
+
+    TS_ASSERT(loaderLOKI.isExecuted());
+
+    // Get back the saved workspace
+    MatrixWorkspace_sptr outputWs;
+    TS_ASSERT_THROWS_NOTHING(
+        outputWs = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
```

Comment:
As above mention. Just:
```
MatrixWorkspace_sptr outputWs =loaderLOKI. getProperty("OutputWorkspace")
```
unfortunately for reasons I won't get into you also need to:
```
loaderLOKI. setPropertyValue("OutputWorkspace", "some_dummy_value"); // BEFORE execute
```

Reply:
Again, copied from NIMROD test. Should I remove it also in there?

Reply:
I'll leave that up to you, but for you new code, do it in the suggested way.